lib/datasource.js

```diff
@@ -192,20 +193,22 @@ function Datasource (key) {
 
   // streaming check if the entry is a metadata file, and if so
   // that it has been downloaded
-  self.ensureMetaFile = through.ctor({ objectMode: true }, (entry, enc, next) => {
-    if (/^meta.*json$/.test(entry.name)) {
-      if (self.archive.isEntryDownloaded(entry)) {
-        next(null, self.loadEntry(entry))
-      } else {
-        self.archive.download(entry, err => {
-          if (err) return next(err)
+  self.ensureMetaFile = function () {
+    return parallel(128, (entry, next) => {
```

Comment:
128 is a magic number indicating how many of these should run in parallel. Feel free to change it (I like powers of two because I'm a computer person). parallel-transform makes sure to preserve the order of the stream so everything should still work